tests/accuracy/test_harpy.py

```diff
@@ -124,14 +149,17 @@ def _get_model_dataframe():
 
 def _write_tbt_file(model, dir_path):
     ints = np.arange(NTURNS) - NTURNS / 2
-    data_x = model.loc[:, "AMPX"].values[:, None] * np.cos(
-        2 * np.pi * (model.loc[:, "MUX"].values[:, None] +
-                     model.loc[:, "TUNEX"].values[:, None] * ints[None, :]))
-    data_y = model.loc[:, "AMPY"].values[:, None] * np.cos(
-        2 * np.pi * (model.loc[:, "MUY"].values[:, None] +
-                     model.loc[:, "TUNEY"].values[:, None] * ints[None, :]))
+    data_x = model.loc[:, "AMPX"].to_numpy()[:, None] * np.cos(
+        2 * np.pi * (model.loc[:, "MUX"].to_numpy()[:, None] +
+                     model.loc[:, "TUNEX"].to_numpy()[:, None] * ints[None, :]))
+    data_y = model.loc[:, "AMPY"].to_numpy()[:, None] * np.cos(
+        2 * np.pi * (model.loc[:, "MUY"].to_numpy()[:, None] +
+                     model.loc[:, "TUNEY"].to_numpy()[:, None] * ints[None, :]))
+    data_z = AMPZ * BASEAMP * np.ones((NBPMS, 1)) * np.cos(
+        2 * np.pi * (MUZ * np.ones((NBPMS, 1)) +
+                     TUNEZ * np.ones((NBPMS, 1)) * ints[None, :]))
     mats = dict(X=pd.DataFrame(data=np.random.randn(model.index.size, NTURNS) * NOISE + data_x
-                               + COUPLING * data_y, index=model.index),
+                               + COUPLING * data_y + data_z, index=model.index),
                 Y=pd.DataFrame(data=np.random.randn(model.index.size, NTURNS) * NOISE + data_y
                                + COUPLING * data_x, index=model.index))
```

Comment:
why is data_z only added in the X case?

Reply:
that relates to dispersion, which we often have only in horizontal plane...for the tests it does not really matter

Reply:
then probably a DISP factor similar to COUPLING would make it easier understandable

Reply:
there is an AMPZ factor...as the factor should physically relate to D_x * dp/p